specification/iothub/resource-manager/Microsoft.Devices/stable/2023-06-30/examples/iothub_addRoutingCosmosDBEndpoint.json

```diff
@@ -0,0 +1,228 @@
+        "storageEndpoints": {
+          "$default": {
+            "sasTtlAsIso8601": "PT1H",
+            "connectionString": "",
+            "containerName": ""
+          }
+        },
+        "messagingEndpoints": {
+          "fileNotifications": {
+            "lockDurationAsIso8601": "PT1M",
+            "ttlAsIso8601": "PT1H",
+            "maxDeliveryCount": 10
+          }
+        },
+        "enableFileUploadNotifications": false,
+        "cloudToDevice": {
+          "maxDeliveryCount": 10,
+          "defaultTtlAsIso8601": "PT1H",
+          "feedback": {
+            "lockDurationAsIso8601": "PT1M",
+            "ttlAsIso8601": "PT1H",
+            "maxDeliveryCount": 10
+          }
+        },
+        "features": "None",
+        "minTlsVersion": "1.2",
+        "enableDataResidency": false
+      },
+      "sku": {
+        "name": "S1",
+        "capacity": 1
+      }
+    }
+  },
+  "responses": {
+    "201": {
+      "body": {
+        "id": "/subscriptions/ae24ff83-d2ca-4fc8-9717-05dae4bba489/resourceGroups/myResourceGroup/providers/Microsoft.Devices/IotHubs/testHub",
+        "name": "testHub",
+        "type": "Microsoft.Devices/IotHubs",
+        "location": "centraluseuap",
+        "tags": {},
+        "etag": "AAAAAAFD6M4=",
+        "properties": {
+          "state": "Active",
+          "provisioningState": "Succeeded",
+          "ipFilterRules": [
+            {
+              "filterName": "rule1",
+              "action": "Accept",
+              "ipMask": "131.117.159.53"
+            },
+            {
+              "filterName": "rule2",
+              "action": "Accept",
+              "ipMask": "157.55.59.128/25"
+            }
+          ],
+          "networkRuleSets": {
+            "defaultAction": "Deny",
+            "applyToBuiltInEventHubEndpoint": true,
+            "ipRules": [
+              {
+                "filterName": "rule1",
+                "action": "Allow",
+                "ipMask": "131.117.159.53"
+              },
+              {
+                "filterName": "rule2",
+                "action": "Allow",
+                "ipMask": "157.55.59.128/25"
+              }
+            ]
+          },
+          "hostName": "iot-dps-cit-hub-1.azure-devices.net",
+          "eventHubEndpoints": {
+            "events": {
+              "retentionTimeInDays": 1,
+              "partitionCount": 2,
+              "partitionIds": [
+                "0",
+                "1"
+              ],
+              "path": "iot-dps-cit-hub-1",
+              "endpoint": "sb://iothub-ns-iot-dps-ci-245306-76aca8e13b.servicebus.windows.net/"
+            }
+          },
+          "routing": {
+            "endpoints": {
+              "serviceBusQueues": [],
+              "serviceBusTopics": [],
+              "eventHubs": [],
+              "storageContainers": [],
+              "cosmosDBSqlContainers": [
+                {
+                  "id": "009208fa-9cb7-4b90-bfbd-f40086e766c9",
```

Comment:
>  "id": "009208fa-9cb7-4b90-bfbd-f40086e766c9", [](http://example.com/codeflow?start=16&length=47)

In ARM APIs 'id' often means the resource id. I suppose you have to do it this way for consistency, but if not, I would have wondered if we could find a better name, maybe uuid or uniqueId, or if it is even really needed. Is there any way the customer will consume this by passing it to other parts of the API?

Reply:
id is inherited name as in all other endpoints